feat(curriculum): Add questions, answers & distractors for javascript date-quiz (#56754)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/quiz-javascript-dates/66edd3711bb9f7efa73aef91.md

#### --distractors--

`Date.new()`

---

`Date.fetch()`

---

`new.Date()`

#### --answer--

`new Date()`

### --question--

#### --text--

How does `Date.now()` work?

#### --distractors--

It returns the current time in nanoseconds since July 4, 1776.

---

It returns the previous time in milliseconds since January 1, 1974.

---

It returns the current date in minutes since January 1, 1970.

#### --answer--

It returns the current time in milliseconds since January 1, 1970.

### --question--

#### --text--

To get the current year we use which instance method of the `Date` object?

#### --distractors--

`getNewYear()`

---

`getLastYear()`

---

`getLeapYear()`

#### --answer--

`getFullYear()`

### --question--

#### --text--

To get the current month(0-11) we use which instance method of the `Date` object?

#### --distractors--

`getCalender()`

---

`fetchMonth()`

---

`get.Month()`

#### --answer--

`getMonth()`

### --question--

#### --text--

The built in javascript method that allows us to format a date as a string is?

#### --distractors--

`getMonth()`

---

`formatDate()`

---

`toStingy()`

#### --answer--

`toString()`

### --question--

#### --text--

> It is the year 2022 this code `console.log(new Date().getFullYear());` outputs what ?

Author: Yes the answer is a bit glaring but its just to make the concept stick better, like if it is the year 2022, what would `.getFullYear()` output ?

Just something easy to prepare them for tougher ones below. I think `: )`

#### --distractors--

> June

Reviewer: ```suggestion
`"June"`
```

---

> 1960

Reviewer: ```suggestion
`1960`
```

---

> 2008

Reviewer: ```suggestion
`2008`
```

#### --answer--

> 2022

Reviewer: ```suggestion
`2022`
```

### --question--

#### --text--

> If today's date was 18th of June, 2023. Which line of code below would give us the output `6/18/2023`

Reviewer: ```suggestion
If today's date was 18th of June, 2023. Which line of code below would give us the output `"6/18/2023"`?
```

#### --distractors--

`console.log('new Date().toLocaleDateString()');`

---

`console.log(new Date().toCityDateString());`

---

`console.log(new Date().toLocaleTimeString());`

#### --answer--

> `console.log(new Date().toLocaleDateString());`

Author: can we refine the question to this

### --question--

#### --text--

What will be the output of the following code?

```js
console.log(new Date("2023-06-18T12:00:00").toUTCString());
```

#### --distractors--

`Mon, 19 Jun 2023 12:00:00 GMT`

---

`Sun, 18 Jun 2023 12:00:00 GMT+1`

---

`Sun, 18 Jun 2023 13:00:00 GMT`

#### --answer--

`Sun, 18 Jun 2023 12:00:00 GMT`

### --question--

#### --text--

If the current time was 1:00pm. Which line of code below would give us the output `1:00:00 PM`

#### --distractors--

`console.log(new Date().toLocaleDateString());`

---

`console.log(new Date().toString());`

---

`console.log(new Date().toCityDateString());`

#### --answer--

> `console.log(new Date().toLocaleTimeString());`

Author: what if we refine the question to this?

### --question--

#### --text--

If the time in your locale is formatted as `HH:MM:SS AM/PM`, which line of code would correctly display the current time?

#### --distractors--

`console.log(new Date().toLocaleDateString());`

---

`console.log(new Date().toString());`

---

`console.log(new Date().toCityDateString());`

#### --answer--

`console.log(new Date().toLocaleTimeString());`

### --question--

#### --text--

The `toISOString()` method formats a date as?

#### --distractors--

A USA-863 string format.

---

> An MP3 string format

Reviewer: ```suggestion
An MP3 string format.
```

---

> A UTC string format

Reviewer: ```suggestion
A UTC string format.
```

#### --answer--

An ISO-8601 string format.

### --question--

#### --text--

What should an ISO-8601 date format look like?

#### --distractors--

> DAY-MONTH-YEAR

Reviewer: ```suggestion
DD-MM-YYYY
```

---

> DAY-MONTH-YEAR:HOUR

Reviewer: ```suggestion
DD-MM-YYTHH
```

---

> DAY-MONTH-YEAR HOUR:MINUTES:SECONDS

Reviewer: ```suggestion
YYYY-MM-DDTHH:mm:ssZ
```

#### --answer--

> YEAR-MONTH-DAY HOUR:MINUTES:SECONDS:MILLISECONDS

Reviewer: ```suggestion
YYYY-MM-DDTHH:mm:ss.sssZ
```

### --question--

#### --text--

What is the output of `console.log(new Date(2003, 6, 27).getMonth());`?

#### --distractors--

> 27/6/2003

Reviewer: ```suggestion
`"27/6/2003"`
```

---

> 27

Reviewer: ```suggestion
`27`
```

---

> 2003

Reviewer: ```suggestion
`2003`
```

#### --answer--

> 6

Reviewer: ```suggestion
`6`
```

### --question--

#### --text--

How would you format a date to a locale-specific string or a more readable format?

#### --distractors--

`.toLocaleDate()`

---

`.toLocaleString()`

---

`.toLocaleTimetring()`

#### --answer--

`.toLocaleDateString()`

### --question--

#### --text--

The `.toLocaleDateString()` date format looks like?

#### --distractors--

MONTH / DAY / SECONDS

---

MONTH / DAY

---

YEAR / MONTH / DAY

#### --answer--

MONTH / DAY / YEAR

### --question--

#### --text--

What does `new Date().getDay()` return?

#### --distractors--

It returns the current day of the week as a string

---

> It returns the current day of the year as a boolean.

Reviewer: ```suggestion
It returns the current day of the year as a number. Starting with Monday as `0`.
```

---

It returns the current day of the month as a decimal.

#### --answer--

> It returns the current day of the week as a number.

Reviewer: ```suggestion
It returns the current day of the week as a number. Starting with Sunday as `0`.
```

### --question--

#### --text--

What is the output of `console.log(new Date(2003, 6, 27).getFullYear());`?

#### --distractors--

> 27/6/2003

Reviewer: ```suggestion
`"27/6/2003"`
```

---

> 27

Reviewer: ```suggestion
`27`
```

---

> 6

Reviewer: ```suggestion
`6`
```

#### --answer--

> 2003

Reviewer: ```suggestion
`2003`
```

### --question--

#### --text--

The `toUTCString()` method formats a date as?

#### --distractors--

A door dash string format

---

An MP3 string format

---

An ISO-8601 string format

#### --answer--

A UTC string format

### --question--

#### --text--

If we are in the month of October this code `console.log(new Date().getMonth());` outputs what ?

#### --distractors--

> 1

Reviewer: ```suggestion
`1`
```

---

> 3

Reviewer: ```suggestion
`3`
```

---

> 10

Reviewer: ```suggestion
`10`
```

#### --answer--

> 9

Reviewer: ```suggestion
`9`
```

### --question--

#### --text--

> If the ouput of this code is "2021", what date method is missing from `console.log(new Date("2021-12-25")(MISSING METHOD))`;

Reviewer: ````suggestion
Which option will output `"2021"` for the following object?

```js
const d = new Date("2021-12-25");
```
````
I think something like this will make it easier to understand the question.

Author: Yes it would

#### --distractors--

> `.getTime()`

Reviewer: ```suggestion
`console.log(d.getTime())`
```

---

> `.toUTCString()`

Reviewer: ```suggestion
`console.log(d.toUTCString())`
```

---

> `.getDate()`

Reviewer: ```suggestion
`console.log(d.getDate())`
```

#### --answer--

> `.getFullYear()`

Reviewer: ```suggestion
`console.log(d.getFullYear())`
```

### --question--

#### --text--

> If the ouput of this code is "2021-12-25T00:00:00.000Z", what date method is missing from `console.log(new Date("2021-12-25")(MISSING METHOD))`;

Reviewer: ````suggestion
Which option will output `"2021-12-25T00:00:00.000Z"` for the following object?

```js
const d = new Date("2021-12-25");
```
````

#### --distractors--

> `.getFullYear()`

Reviewer: ```suggestion
`console.log(d.getFullYear())`
```

---

> `.toUTCString()`

Reviewer: ```suggestion
`console.log(d.toUTCString())`
```

---

> `.getTime()`

Reviewer: ```suggestion
`console.log(d.getTime())`
```

#### --answer--

> `.toISOString()`

Reviewer: ```suggestion
`console.log(d.toISOString())`
```

#### --text--

The `Date` object in javascript is used to?

#### --distractors--

Work with just dates only.

---

Work with just time only.

---

> None of the above.

Reviewer 2: ```suggestion
None of the others.
```

the choices are in random order

Author: Thank you. Its been corrected,

#### --distractors--

`Date.new()`

---

`Date.fetch()`

---

`new.Date()`

#### --answer--

`new Date()`

### --question--

#### --text--

How does `Date.now()` work?

#### --distractors--

It returns the current time in nanoseconds since July 4, 1776.

---

It returns the previous time in milliseconds since January 1, 1974.

---

It returns the current date in minutes since January 1, 1970.

#### --answer--

It returns the current time in milliseconds since January 1, 1970.

### --question--

#### --text--

To get the current year we use which instance method of the `Date` object?

#### --distractors--

`getNewYear()`

---

`getLastYear()`

---

`getLeapYear()`

#### --answer--

`getFullYear()`

### --question--

#### --text--

To get the current month (0 - 11) we use which instance method of the `Date` object?

#### --distractors--

`getCalender()`

---

`fetchMonth()`

---

`get.Month()`

#### --answer--

`getMonth()`

### --question--

#### --text--

The built-in JavaScript method that allows us to format a date as a string is?

#### --distractors--

`getMonth()`

---

`formatDate()`

---

`toStingy()`

#### --answer--

`toString()`

### --question--

#### --text--

> What would output `console.log(new Date().getFullYear());`, if it would be the year `2022`?

Reviewer 2: ```suggestion
What would output `console.log(new Date().getFullYear());`, if it would be June 12th 2022?
```

so that the June answer makes sense. One of the other two distractors could maybe be some other version of this date?

Author: Yes I feel that would make more sense than just giving out the answer. Let me quickly run the changes.

Author: Do this distractors make sense, I hope it was in line with your suggestion ?

### --question--

#### --text--

What would output `console.log(new Date().getFullYear());`, if it would be June 12th 2022?

#### --distractors--

`"June 12th 2022"`

---

`2023`

---

`06/12/2022`

#### --answer--

`2022`

Reviewer 2: They make sense even before. But if one of the distractor it's a month it makes sense to have it in the question date too

Author: Oh Yes I see

#### --distractors--

`"June 12th 2022"`

---

`2023`

---

`06/12/2022`

#### --answer--

`2022`

### --question--

#### --text--

What will be the output of the following code?

```js

> console.log(new Date("2023-06-18T12:00:00").toUTCString());

Author: hmmm your right, without adding the z at the end of the date string when it is converted to UTC depending on the users time zone, the hours would get subtracted. 

Like for example i was getting
Sun, 18 Jun 2023 11:00:00 GMT this.

but  

```js
console.log(new Date("2023-06-18T12:00:00Z").toUTCString());
````
adding z seemed to fix it. Now the date object interprets the provided time as UTC without applying any local time zone offset by subtracting.

Giving the answer as `Sun, 18 Jun 2023 12:00:00 GMT` no matter what the users time locale is.

#### --distractors--

`console.log(new Date().toLocaleDateString());`

---

`console.log(new Date().toString());`

---

`console.log(new Date().toCityDateString());`

#### --answer--

`console.log(new Date().toLocaleTimeString());`

### --question--

#### --text--

The `toISOString()` method formats a date as?

#### --distractors--

A USA-863 string format.

---

An MP3 string format.

---

A UTC string format.

#### --answer--

An ISO-8601 string format.

### --question--

#### --text--

What should an ISO-8601 date format look like?

#### --distractors--

DD-MM-YYYY

---

DD-MM-YYTHH

---

YYYY-MM-DDTHH:mm:ssZ

#### --answer--

YYYY-MM-DDTHH:mm:ss.sssZ

### --question--

#### --text--

What is the output of `console.log(new Date(2003, 6, 27).getMonth());`?

#### --distractors--

`"27/6/2003"`

---

`27`

---

`2003`

#### --answer--

`6`

### --question--

#### --text--

How would you format a date to a locale-specific string or a more readable format?

#### --distractors--

`.toLocaleDate()`

---

`.toLocaleString()`

---

`.toLocaleTimetring()`

#### --answer--

`.toLocaleDateString()`

### --question--

#### --text--

> Which option will correctly create a `Date` object representing January 1, 2025?

Reviewer: ```
new Date("1-1-2025")
Wed Jan 01 2025 00:00:00 GMT+0100
new Date(2025, 0, 0);
Tue Dec 31 2024 00:00:00 GMT+0100
new Date("2025-01-01T00:00:00");
Wed Jan 01 2025 00:00:00 GMT+0100
new Date("2025-01-01")
Wed Jan 01 2025 01:00:00 GMT+0100
```
This isn't wrong (remember about local time-zone), but I worry it can be very confusing.

Author: I see what you mean by confusing, what if we format the question to this, what do u think ? This would remove the problem or issue we keep having with the users different time zone.

### --question--

#### --text--

Which option will correctly create a string representing January 1, 2025?

```js
const d = ?;
console.log(d.toISOString().split("T")[0]);
```

#### --distractors--

`const d = new Date("2021-02-01");`

---

`const d = new Date("2003-06-06");`

---

`const d = new Date("2001-04-08");`

#### --answer--

`const d = new Date("2025-01-01");`

#### --text--

What is the JavaScript `Date` object used for?

#### --distractors--

Work with just dates only.

---

Work with just time only.

---

None of the others.

#### --answer--

Work with both dates and time.

### --question--

#### --text--

Which code creates a new `Date` object instance?

#### --distractors--

`Date.new()`

---

`Date.fetch()`

---

`new.Date()`

#### --answer--

`new Date()`

### --question--

#### --text--

How does `Date.now()` work?

#### --distractors--

> It returns the current time in nanoseconds since July 4, 1776.

Reviewer 3: I wonder how many Americans we'll trap with this one. 🤣
